File: examples/FSharp/FSharp.Examples/Scenarios/DataFeed.fs
module DataFeedScenario

open System
open System.Threading.Tasks

open FSharp.Control.Tasks.V2.ContextInsensitive

open NBomber
open NBomber.Contracts
open NBomber.FSharp

[<CLIMutable>]
type User = {
    Id: int
    Name: string
}

let run () =

    let data = FeedData.fromSeq [1; 2; 3; 4; 5]
               |> FeedData.shuffleData

    //let data = FeedData.fromJson<User>("users_feed_data.json")
    //let data = FeedData.fromCsv<User>("users_feed_data.csv")

    let feed = Feed.createConstant "numbers" data
    //let feed = Feed.createConstant "numbers" data
    //let feed = Feed.createRandom "numbers" data

    let step = Step.create("simple step", feed, fun context -> task {

        do! Task.Delay(TimeSpan.FromSeconds 1.0)

        context.Logger.Information("Data from feed: {FeedItem}", context.FeedItem)
        return Response.Ok()
    })

    let scenario = Scenario.create "Hello World!" [step]

    NBomberRunner.registerScenarios [scenario]
    |> NBomberRunner.run
    |> ignore

File: examples/CSharp/CSharp.Examples/Scenarios/DataFeed.cs
using System;
using System.Threading.Tasks;

using NBomber;
using NBomber.Contracts;
using NBomber.CSharp;

namespace CSharp.Examples
{
    class DataFeedScenario
    {
        public static void Run()
        {
            var data = FeedData.FromSeq(new[] {1, 2, 3, 4, 5}).ShuffleData();
            //var data = FeedData.FromJson<User>("users_feed_data.json");
            //var data = FeedData.FromCsv<User>("users_feed_data.csv");

            var feed = Feed.CreateCircular("numbers", data);
            //var feed = Feed.CreateConstant("numbers", data);
            //var feed = Feed.CreateRandom("numbers", data);

            var step = Step.Create("step", feed, async context =>
            {
                await Task.Delay(TimeSpan.FromSeconds(1));

                context.Logger.Information("Data from feed: {FeedItem}", context.FeedItem);
                return Response.Ok();
            });

            var scenario = ScenarioBuilder.CreateScenario("Hello World!", new[] {step});

            NBomberRunner
                .RegisterScenarios(new[] {scenario})
                .RunInConsole();
        }

        public class User
        {
            public long Id { get; set; }
            public string Name { get; set; }
        }
    }
}
